### ISSUE TITLE
Promise not run `resolve` or `reject` when config file not exist.

### ISSUE THREAD

**Issue by gucong3000** [davidtheclark/cosmiconfig#28]
Windows can reproduce the bug

```javascript
return stylelint.getConfigForFile("E:\\work\\stylefmt\\test\\fixtures\\at-apply\\at-apply.css")

.then(config => {
  // not run when configfile not exist.
  console.log(config)
})

.catch(ex => {
  // not run when configfile not exist.
  console.error(ex)
})
```

https://github.com/davidtheclark/cosmiconfig/blob/master/lib/createExplorer.js#L79
Always `false` in Windows. It is infinite loop.

**Comment by davidtheclark** [comment]
@gucong3000 Does this bug only happen when you pass windows-style path strings? What if you pass unix-style path strings, or use Node's `path.join` to create the path?

**Comment by davidtheclark** [comment]
Also, that's not a valid Windows path, right? Why the double-slashes? What happens if you make them single slashes — does it work?

**Comment by gucong3000** [comment]
@davidtheclark 

Valid path in Windows can reproduce the bug.

> Does this bug only happen when you pass windows-style path strings?

Yes.

> What if you pass unix-style path strings, or use Node's path.join to create the path?

If not run in Windows, It is work properly.


> that's not a valid Windows path, right?

that's valid Windows path.

> Why the double-slashes?

Escape character. please try `console.log("E:\\")`, this will output single slash, like `E:\`

**Comment by gucong3000** [comment]
```javascript
test.serial('Configuration file not exist', (assert) => {
  const loadConfig = cosmiconfig('not_exist_rc_name').load;
  return loadConfig('.').then((result) => {
    assert.deepEqual(result, null);
  });
});
```
Running this under Windows , the states of Promise will always be "pending""

**Comment by gucong3000** [comment]
There is tow way can fix.

1. `splitPath()` only run once.
2. do not use `path.resolve()` in `splitPath()`

[Specific reasons for bug production](https://github.com/davidtheclark/cosmiconfig/pull/31#discussion_r82718656)

**Comment by davidtheclark** [comment]
Closed by #29
